### PR TITLE
feat: Publishing Without RELEASES.md PR mode

### PR DESCRIPTION
**Pull request by ryan-timothy-albert**
Allows for publishing without reading from a RELEASES.md file in PR mode

Related sdk-gen-config PR: https://github.com/speakeasy-api/sdk-gen-config/pull/59
RFC: https://www.notion.so/speakeasyapi/RFC-Generation-Action-Independent-Publishing-113726c497cc8055812ef25ea0cc7bb8

A few notes
- We need to leave around the RELEASES.md trigger for backward compatibility with existing actions
- New actions will have two triggers
    -  change to `gen.lock` on branch
    - workflow dispatch (on demand)
- this only modified the publishing flow in pr mode right now. I want to slowly stage changes given the sensitivity of it. Adding similar support in direct-mode (not relying on RELEASES.md) will be a small change
- Generally I just need to do a ton of testing for a bunch of different usecases before I actually merge this.

Demo: https://www.loom.com/share/ce382b588ed6415590b815f1363c4bd6?sid=1764be17-2bf8-4c14-9f4a-47274fe50ec1

Later on we will look to move writing of RELEASES.md into the CLI

### CODE REVIEW

File: .github/workflows/sdk-publish.yaml [hunk 1]
```diff
@@ -16,6 +16,10 @@ on:
         required: false
         type: string
         default: "5.x"
+      target:
+        description: "Generate a specific target by name"
```
Comment (chase-crumbaugh) [inline .github/workflows/sdk-publish.yaml:20]: Not sure the description even appears anywhere, but still, this doesn't seem right

File: internal/actions/release.go [hunk 2]
```diff
@@ -25,30 +26,64 @@ func Release() error {
...
 	}
 
-	files, err := g.GetCommitedFiles()
-	if err != nil {
-		fmt.Printf("Failed to get commited files: %s\n", err.Error())
+	dir := "."
+	usingReleasesMd := false
+	// TODO: Maybe instead check for workflow dispatch event
+	providesExplicitTarget := environment.SpecifiedTarget() != ""
+	if providesExplicitTarget {
```
Comment (chase-crumbaugh) [inline internal/actions/release.go:33]: Love this english-language code 🙌

File: internal/actions/release.go [hunk 3]
```diff
@@ -25,30 +26,64 @@ func Release() error {
 		return err
 	}
 
-	files, err := g.GetCommitedFiles()
-	if err != nil {
-		fmt.Printf("Failed to get commited files: %s\n", err.Error())
+	dir := "."
+	usingReleasesMd := false
+	// TODO: Maybe instead check for workflow dispatch event
```
Comment (chase-crumbaugh) [inline internal/actions/release.go:31]: Could it make sense to provide workflow dispatch event without a particular target?

File: internal/actions/release.go [hunk 4]
```diff
@@ -25,30 +26,64 @@ func Release() error {
...
 
-	for _, file := range files {
-		if strings.Contains(file, "RELEASES.md") {
-			dir = filepath.Dir(file)
-			logging.Info("Found RELEASES.md in %s\n", dir)
-			break
+			if strings.Contains(file, "gen.lock") {
+				dir = filepath.Dir(file)
+				dir = strings.ReplaceAll(dir, "/.speakeasy", "")
+				dir = strings.ReplaceAll(dir, ".speakeasy", "")
```
Comment (chase-crumbaugh) [inline internal/actions/release.go:69]: I don't fully understand why we're dropping the .speakeasy part of the directory

**Review by chase-crumbaugh — APPROVED** [review]

File: internal/actions/release.go (hunk 3 above)
Comment (ryan-timothy-albert) [reply internal/actions/release.go:31]: Not at this point. Publishing files will always be coded with a specific target now

File: internal/actions/release.go (hunk 4 above)
Comment (ryan-timothy-albert) [reply internal/actions/release.go:69]: Because we want the path to the actual SDK

**Pull request merged**